test/src/updater/differentialUpdateTest.ts

```diff
@@ -171,22 +171,25 @@ test.ifAll.ifDevOrLinuxCi("AppImage", () => testLinux(Arch.x64))
 
 test.ifAll.ifDevOrLinuxCi("AppImage ia32", () => testLinux(Arch.ia32))
 
-// ifAll.ifMac.ifNotCi todo
-test.skip("dmg", async () => {
+async function testMac(arch: Arch) {
+  process.env.TEST_UPDATER_ARCH = Arch[arch]
+
   const outDirs: Array<string> = []
   const tmpDir = new TmpDir("differential-updater-test")
-  if (process.env.__SKIP_BUILD == null) {
-    await doBuild(outDirs, Platform.MAC.createTarget(undefined, Arch.x64), tmpDir, {
+  try {
+    await doBuild(outDirs, Platform.MAC.createTarget(["dmg"], arch), tmpDir, {
       mac: {
         electronUpdaterCompatibility: ">=2.17.0",
       },
     })
-  } else {
-    // todo
+    await testBlockMap(outDirs[0], path.join(outDirs[1]), MacUpdater, "mac/Test App ßW.app", Platform.MAC)
+  } finally {
```

Comment:
Could this `try-finally` be hiding if the test fails? As it stands, I don't think `dmg` would work for differential updates, only `zip` target should work based on how the auto-updater serves the update over secure localhost server to the built-in electron's native updater functionality.

>zip target for macOS is required for Squirrel.Mac, otherwise latest-mac.yml cannot be created, which causes autoUpdater error. Default target for macOS is dmg+zip, so there is no need to explicitly specify target.

Ref: https://www.electron.build/auto-update

Reply:
dmg works for upgrading, so I think it also would work for differential updates.

Reply:
zip target for macOS is required for Squirrel.Mac, otherwise latest-mac.yml cannot be created, which causes autoUpdater error. Default target for macOS is dmg+zip, so there is no need to explicitly specify target.(https://www.electron.build/auto-update). So dmg works for auto update.